### ISSUE TITLE
Login errors

### ISSUE THREAD

**Issue by RoyWinters** [ubis/Freya#5]
First of all: Can't get client from wiki page link, xtrap blocks start

**Using client from** 
https://forum.ragezone.com/threads/public-all-available-emulators-of-cabal-online.1216337/#post-9199589
Archive name CABAL_Online_Europe_READ2USE EMU CLIENT
Get this error 
![Untitled](https://github.com/ubis/Freya/assets/18345014/17e7f662-300b-471b-8443-0a0767da4bc7)

loginserver.log
```
2023-08-11 01:57:58.639 [INFO] Client `192.168.50.238:54735` connected to the LoginServer
2023-08-11 01:57:58.639 [DEBUG] Received Packet `Connect2Svr` (Len: 14, type: 101, src: 192.168.50.238:54735)
2023-08-11 01:57:58.639 [DEBUG] Sent Packet `Connect2Svr` (Len: 18, type: 101, src: 192.168.50.238:54735)
2023-08-11 01:57:58.640 [DEBUG] Received Packet `CheckVersion` (Len: 26, type: 122, src: 192.168.50.238:54735)
2023-08-11 01:57:58.640 [ERROR] Client version mismatch (Client: 1421, server: 374, src: 192.168.50.238:54735)
2023-08-11 01:57:58.640 [DEBUG] Sent Packet `CheckVersion` (Len: 22, type: 122, src: 192.168.50.238:54735)
2023-08-11 01:58:01.216 [ERROR] Error reading: read tcp 192.168.50.238:38101->192.168.50.238:54735: wsarecv: An existing connection was forcibly closed by the remote host.
2023-08-11 01:58:01.216 [INFO] Client `192.168.50.238:54735` disconnected from the LoginServer
```

**Using client from** 
https://forum.ragezone.com/threads/ep8-server-in-docker.1202852/

On login get this error 

![image](https://github.com/ubis/Freya/assets/18345014/89db3fea-0ace-40c4-b545-26e86536ba10)

loginserver.log
```
2023-08-11 00:57:31.916 [INFO] Client `192.168.50.238:52347` connected to the LoginServer
2023-08-11 00:57:31.916 [DEBUG] Received Packet `Connect2Svr` (Len: 14, type: 101, src: 192.168.50.238:52347)
2023-08-11 00:57:31.917 [DEBUG] Sent Packet `Connect2Svr` (Len: 18, type: 101, src: 192.168.50.238:52347)
2023-08-11 00:57:31.918 [DEBUG] Received Packet `CheckVersion` (Len: 26, type: 122, src: 192.168.50.238:52347)
2023-08-11 00:57:31.918 [DEBUG] Sent Packet `CheckVersion` (Len: 22, type: 122, src: 192.168.50.238:52347)
2023-08-11 00:57:31.919 [DEBUG] Received Packet `PreServerEnvRequest` (Len: 43, type: 2002, src: 192.168.50.238:52347)
2023-08-11 00:57:31.919 [DEBUG] Sent Packet `PreServerEnvRequest` (Len: 4119, type: 2002, src: 192.168.50.238:52347)
2023-08-11 00:57:31.920 [DEBUG] Received Packet `PublicKey` (Len: 10, type: 2001, src: 192.168.50.238:52347)
2023-08-11 00:57:31.920 [DEBUG] Sent Packet `PublicKey` (Len: 279, type: 2001, src: 192.168.50.238:52347)
2023-08-11 00:57:31.921 [DEBUG] Received Packet `AuthAccount` (Len: 268, type: 103, src: 192.168.50.238:52347)
2023-08-11 00:57:32.100 [INFO] User `root` successfully logged in.
2023-08-11 00:57:32.100 [DEBUG] Sent Packet `AuthAccount` (Len: 46, type: 103, src: 192.168.50.238:52347)
2023-08-11 00:57:32.100 [DEBUG] Sent Packet `URLToClient` (Len: 173, type: 128, src: 192.168.50.238:52347)
2023-08-11 00:57:32.100 [DEBUG] Sent Packet `SystemMessg` (Len: 52, type: 120, src: 192.168.50.238:52347)
2023-08-11 00:57:32.100 [DEBUG] Sent Packet `SystemMessg` (Len: 46, type: 120, src: 192.168.50.238:52347)
2023-08-11 00:57:32.100 [DEBUG] Sent Packet `SystemMessg` (Len: 9, type: 120, src: 192.168.50.238:52347)
2023-08-11 00:57:32.100 [WARNING] Panic! Recovered from: AuthAccount, src: 192.168.50.238:52347, id: 2
2023-08-11 00:57:32.106 [ERROR] runtime error: slice bounds out of range [:6] with length 0
2023-08-11 00:57:32.107 [ERROR] goroutine 82 [running]:
runtime/debug.Stack()
	C:/Program Files/Go/src/runtime/debug/stack.go:24 +0x5e
github.com/ubis/Freya/share/network.(*PacketHandler).Handle.func1()
	D:/Cabal/Freya-development/share/network/packethandler.go:47 +0x265
panic({0xb237a0?, 0xc00008c120?})
	C:/Program Files/Go/src/runtime/panic.go:914 +0x21f
github.com/ubis/Freya/cmd/loginserver/packet.NotifyServerInfo(0xc000482090?)
	D:/Cabal/Freya-development/cmd/loginserver/packet/packet.go:41 +0xb9
github.com/ubis/Freya/cmd/loginserver/packet.AuthAccount(0xc000482090, 0xc0003e43f0?)
	D:/Cabal/Freya-development/cmd/loginserver/packet/auth.go:94 +0xb4b
github.com/ubis/Freya/share/network.(*PacketHandler).Handle(0xc7d8e0, 0xc000500080)
	D:/Cabal/Freya-development/share/network/packethandler.go:74 +0x102
main.OnPacketReceive({0xaf58c0?, 0xc000500080?})
	D:/Cabal/Freya-development/cmd/loginserver/event.go:50 +0x226
created by github.com/ubis/Freya/share/event.Trigger in goroutine 48
	D:/Cabal/Freya-development/share/event/event.go:34 +0xd1
2023-08-11 00:57:32.108 [ERROR] Error reading: read tcp 192.168.50.238:38101->192.168.50.238:52347: use of closed network connection
2023-08-11 00:57:32.108 [INFO] Client `192.168.50.238:52347` disconnected from the LoginServer
2023-08-11 00:57:32.108 [INFO] Client `192.168.50.238:52347` disconnected from the LoginServer
```

TL;DR
First time creating/emulating server for nostalgics, was playing it in 2009-2011

Cant login with emulator that 
"Official EP8 EU Client + unpacked working Cabalmain
FOR EMULATORS ONLY"

**Comment by ubis** [comment]
Hi,

first one is because client does not match what should be in config:

> Client version mismatch (Client: 1421, server: 374, src: 192.168.50.238:54735)

This is because default login server configuration is set to "TGCabalEP8" client.

You can use these for EU client:
```
[client]
; client version to accept
client_version=1421

; client magic key to accept
magic_key=709182467
```

Second one is because of not-available build commit. Have you built it yourself?

Can you try to use the one from artifacts? Here: https://github.com/ubis/Freya/actions/runs/5813158704

I will push a fix for build with non-git environment.

**Comment by RoyWinters** [comment]
> Hi,
> 
> first one is because client does not match what should be in config:
> 
> > Client version mismatch (Client: 1421, server: 374, src: 192.168.50.238:54735)
> 
> This is because default login server configuration is set to "TGCabalEP8" client.
> 
> You can use these for EU client:
> 
> ```
> [client]
> ; client version to accept
> client_version=1421
> 
> ; client magic key to accept
> magic_key=709182467
> ```

Set client version and magic key to EU client and get same error as second client

> 
> Second one is because of not-available build commit. Have you built it yourself?
> 
> Can you try to use the one from artifacts? Here: https://github.com/ubis/Freya/actions/runs/5813158704
> 
> I will push a fix for build with non-git environment.

I build it to .exe because of windows and this what I find on internet how to launch GO builded files
![image](https://github.com/ubis/Freya/assets/18345014/f2c2e8a5-d770-4f23-a860-f7ab3839fab6)

Same with artifact files on windows "freya-windows". Can't get it launch because of not ".exe", add .exe in the end and IT'S WORKS!

![image](https://github.com/ubis/Freya/assets/18345014/51b18758-d29e-466f-9456-4a1107947d8c)

One question: No mob spawn, quests and other activities. It's because early stage of developing or I need some manual settings?

**Comment by ubis** [comment]
Yes, it was because of non-git environment, now it's fixed.

> One question: No mob spawn, quests and other activities. It's because early stage of developing or I need some manual settings?

Yes, it's quite early development, you can read about features in wiki [here](https://github.com/ubis/Freya/wiki/Features).

Right now I'm working on a world manager which will also come with mobs in maps.

**Comment by RoyWinters** [comment]
Please do some "buymeacoffee" or something else to contribute to your work. This is great! Going to watch your project grow! Thanks

**Comment by ubis** [comment]
Thanks, yeah I guess I will need to add it.
